lib/waterpig/database-cleaner.rb

```diff
@@ -25,18 +37,57 @@ def with_showing(show, detailed)
     end
   end
 
+  def cleaning_policy(config, metadata)
+    return :no_clean if metadata[config.waterpig_exclude_cleaning_key]
+    unless (explicit_method = metadata[config.waterpig_explicit_cleaning_method_key]).nil?
+      if explicit_method == :reset
+        config.waterpig_database_reset_method
+      else
+        explicit_method
```

Comment:
Look one line down - explicit_method is extracted from the metadata_hash at a configured key - so it will usually be nil. If it's not, it's used twice more in the method.